mailpoet/lib/Entities/DynamicSegmentFilterData.php

```diff
@@ -87,6 +88,35 @@ public function getParam(string $name) {
     return $this->filterData[$name] ?? null;
   }
 
+  public function getStringParam(string $name): string {
```

Comment:
This is left over from my initial approach, and you're right it's not currently used anywhere.

I'm inclined to keep it because I think it would be helpful in getting rid of things like this in our PHPStan config: https://github.com/mailpoet/mailpoet/blob/d55742aebdac050e1af943798ab9d63f8ee3189f/mailpoet/tasks/phpstan/phpstan-7-baseline.neon#L539-L541

Maybe it would be better to wait and handle that all in a separate ticket though. I'm happy to remove this from this PR if you prefer!

Reply:
Thanks for the additional details. We can keep it.